tchannel/tornado/peer.py

```diff
@@ -409,6 +411,17 @@ def _send(self, connection, req):
         with timeout(response_future, req.ttl):
             try:
                 response = yield response_future
+            except StreamClosedError as error:
+                error = NetworkError(
+                    id=req.id,
+                    description=error.message,
+                    tracing=req.tracing,
+                )
+                # event: after_receive_error
+                self.tchannel.event_emitter.fire(
+                    EventType.after_receive_error, req, error,
+                )
+                raise
```

Comment:
It is will be raised again since we have a try catch from the caller to tell whether we should retry based on the exception or not. 

This part will be refined in this task: https://github.com/uber/tchannel-python/issues/285